src/App.jsx

```diff
@@ -20,20 +40,28 @@ const App = () => {
   const addChip = (id) => {
     const newDogs = dogs.map((dog) => {
       if (dog.id === id) {
-        return {
-          ...dog,
-          chip: String(getRndInteger(1000, 9999)),
-        };
+        dog.chip = String(getRndInteger(1000, 9999));
+        axios
+          .patch(`${URL}/${id}/add_chip`)
+          .then(() => setDogs(newDogs))
+          .catch((error) => console.log(error));
       }
       return dog;
     });
-    setDogs(newDogs);
   };
 
   const deleteDog = (id) => {
-    const newDogs = dogs.filter((dog) => dog.id !== id);
-    setDogs(newDogs);
+    axios
```

Comment:
👍 This is the style that the suggestion above is aiming for. Make the API call, then use the result (or lack of error in this case)to update the local state.